File: Lokesh_Saana/848-shifting-letters.py
# https://leetcode.com/problems/shifting-letters/submissions/

class Solution:
    def shiftingLetters(self, s: str, shifts: List[int]) -> str:
        temp = 0
        s = list(s)
        for i in range(len(s)-1, -1, -1):
            temp += shifts[i]
            temp %= 26
            if (temp > ord('z')-ord(s[i])):
                s[i] = chr(ord(s[i])+temp-26)
            else:
                s[i] = chr(ord(s[i])+temp)
        return ''.join(s)

File: Ayush-Gupta/848. Shifting Letters.cpp
class Solution {
public:
    string shiftingLetters(string s, vector<int>& A) {
        int i,n=A.size(); 
        for(i=n-2;i>=0;i--) A[i] = (A[i] + A[i+1]) % 26;
        for(i=0;i<n;i++) s[i] = (s[i] - 'a' + A[i]) % 26 + 'a';
        return s;
    }
};